src/fw_iptables.c

```diff
@@ -327,8 +327,12 @@ iptables_fw_init(void)
 
 	// filter CHAIN_INPUT chain
 
-	// packets coming in on gw_interface jump to CHAIN_TO_ROUTER
-	rc |= nftables_do_command("insert rule ip nds_filter %s iifname \"%s\" counter jump %s", CHAIN_INPUT, gw_interface, CHAIN_TO_ROUTER);
+	// packets coming in on gw_interface jump to CHAIN_TO_ROUTER, unless coming from an external FAS
+	if (strcmp(fas_remoteip, gw_ip) == 0 ) {
+		rc |= nftables_do_command("insert rule ip nds_filter %s iifname \"%s\" counter jump %s", CHAIN_INPUT, gw_interface, CHAIN_TO_ROUTER);
+	} else {
+		rc |= nftables_do_command("insert rule ip nds_filter %s iifname \"%s\" ip saddr !=\"%s\" counter jump %s", CHAIN_INPUT, gw_interface, fas_remoteip, CHAIN_TO_ROUTER);
```

Comment:
This rule will allow only the FAS IP to have access to the router, only if coming from the hotspot/LAN interface. 
Clients are not blocked, they are forwarded to the ndsRTR chain, where their access toward the router is managed, as far as I can tell.

Reply:
> Clients are not blocked, they are forwarded to the ndsRTR chain,

Yes, I understand and yes that is where client access to the router is managed.

However, allowing access to everything on the router, by default, for all types of fas on the local network is not acceptable, as it is not normally required.

**_Level 3 is intended ONLY for use with a fas on a public network._** Making https work correctly on a private network is hacky at best and at worst could be regarded as a MITM attack in terms of captive portal implementation.

Using a configuration option to allow a generalised access from the router to a local network service is a better approach. It would be added at the choice of the installer.
An immediate use for this is use of something like a Pi-Hole server or similar. Numerous local subnet hosted services could be supported.

Reply:
>For a level 1 or level 2 fas, these will also have full access without any requirement or indeed any indication that this is >the case. I would see this as a potential security vulnerability.

>This would only be required if the router initiated a call to a service running on a device in the local subnet eg your use case.
>
>To make this rule a built in default, this PR would have to check if the fasremoteip was on the router subnet and then >insert the rule ( preferably using the new library call) into the ndsRTR chain.

Understood and agreed this would be a cleaner solution. I am thinking about allowing FAS to NDS traffic when:
1. Mode is 3
2. FAS is on the same interface as the NDS (The issue is not the subnet itself, but rather that the traffic is coming from the  gateway interface.
3. Use cstate to allow only ESTABLISHED,RELATED and not NEW, so new connections, initiated from the FAS are not allowed, 

@bluewavenet Is this on the right track?

Reply:
> Is this on the right track?

Commit https://github.com/openNDS/openNDS/pull/529/commits/c62f381374a0b79db6b25bd82e7a9a258f178139 allows access to a local service from the router. This will be available in the upcoming v10.2.0 release so has made this PR somewhat redundant.

My main concern is that **_attempts_** to configure https fas on the local network should not be made. Without numerous potentially dubious "fixups", it will always generate security errors on the client device.

Reply:
>Commit https://github.com/openNDS/openNDS/commit/c62f381374a0b79db6b25bd82e7a9a258f178139 allows access to a local service from the router. This will be available in the upcoming v10.2.0 release so has made this PR somewhat redundant.

Just to be clear, this will make it possible to add arbitrary firewall rules in config?

>My main concern is that attempts to configure https fas on the local network should not be made. Without numerous potentially dubious "fixups", it will always generate security errors on the client device.

I am not sure this should be a concern for OpenNDS. Controlling local DNS and owning a domain should make this trivial and perfectly above board, without generating any security errors and/or compromising trust.

Reply:
The option format is:
`option ruleset '[allow|block] [tcp|udp|all] [port portnumber] [to|from] ipaddress'`

For example:
`option users_to_router 'allow all from 192.168.1.10'`

> I am not sure this should be a concern for OpenNDS

It is a concern. It should be (and will be) reflected in all the documentation but should not be hardwired in a blocked state. It is now configurable and if desired, can be enabled as a deliberate choice.

Reply:
> It is a concern. It should be (and will be) reflected in all the documentation but should not be hardwired in a blocked state. It is now configurable and if desired, can be enabled as a deliberate choice.

I was referring to this:

> Without numerous potentially dubious "fixups", it will always generate security errors on the client device.

Meaning that whether someone knows how to configure SSL on their own local (or remote) FAS, should be neither a concern, nor a feature driver for a project like OpenNDS. 

On the actual point though, while the ruleset option would cover this use case, it would still open up the router to more than is needed, such as the FAS initiating the connection. That is why I was proposing using connection state as a way to further limit access to the router.

If you think that this use case should be treated as a separate configuration option, instead of handled directly by OpenNDS logic, then I guess this PR should be closed.

Reply:
> such as the FAS initiating the connection.

Initiating a connection with what? 
The `'allow all from 192.168.1.10'` (or whatever is the FAS ip) allows responses from FAS back to the router (thus allowing the "nat traversal" tunnel to establish (yes this is intended to allow an Internet hosted FAS to talk to openNDS through the inbound firewall).